source/ControllableHero.hx

```diff
@@ -0,0 +1,49 @@
+import flixel.FlxG;
+import flixel.FlxSprite;
+import flixel.util.FlxColor;
+
+class ControllableHero extends FlxSprite
+{
+	var speed:Int;
+
+	public function new(X:Int = 0, Y:Int = 0, ?speed:Int = 100)
+	{
+		super(X, Y);
+		this.speed = speed;
+		this.makeGraphic(50, 12, FlxColor.WHITE, false);
+		drag.x = this.speed * 10;
+		drag.y = this.speed * 10;
+	}
+
+	function movement()
+	{
+		final left = FlxG.keys.pressed.LEFT;
+		final right = FlxG.keys.pressed.RIGHT;
+		final up = FlxG.keys.pressed.UP;
+		final down = FlxG.keys.pressed.DOWN;
+
+		// this is where I learned how to use velocity
+		if (left)
+		{
+			velocity.x = -speed;
+		}
+		else if (right)
+		{
+			velocity.x = speed;
+		}
+		else if (up)
+		{
+			velocity.y = -speed;
+		}
+		else if (down)
+		{
+			velocity.y = speed;
+		}
```

Comment:
Worth noting that this system of checking inputs creates a "priority order" sort of system - if left is held, all other inputs will be ignored and so on. Might not be desirable long-term for a game, but functional for our purposes.